google-api-client/src/main/java/com/google/api/client/googleapis/apache/v2/GoogleApacheHttpTransport.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.api.client.googleapis.apache.v2;
+
+import com.google.api.client.googleapis.GoogleUtils;
+import com.google.api.client.http.apache.v2.ApacheHttpTransport;
+import com.google.api.client.util.SslUtils;
+import java.io.IOException;
+import java.net.ProxySelector;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.util.concurrent.TimeUnit;
+import javax.net.ssl.SSLContext;
+import org.apache.http.client.HttpClient;
+import org.apache.http.config.SocketConfig;
+import org.apache.http.conn.socket.LayeredConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.impl.conn.SystemDefaultRoutePlanner;
+
+/**
+ * Utilities for Google APIs based on {@link ApacheHttpTransport}.
+ *
+ * @since 1.31
+ */
+public final class GoogleApacheHttpTransport {
+
+  /**
+   * Returns a new instance of {@link ApacheHttpTransport} that uses
+   * {@link GoogleUtils#getCertificateTrustStore()} for the trusted certificates.
+   */
+  public static ApacheHttpTransport newTrustedTransport() throws GeneralSecurityException,
+      IOException {
+    // Set socket buffer sizes to 8192
+    SocketConfig socketConfig =
```

Comment:
DANGER! This can cause major performance issues, and indeed we saw this with Jib or Skaffold a year or two back, and it was really hard to debug. Just accept the defaults.

Reply:
see https://github.com/googleapis/google-http-java-client/pull/855

Reply:
Removed. This was taken verbatim from the original `GoogleApacheHttpTransport` implementation.